src/auth/auth.proto

```diff
@@ -156,27 +112,58 @@ message AuthenticateResponse {
 message WhoAmIRequest {}
 
 message WhoAmIResponse {
+  reserved 2,4;
   string username = 1;
-  bool is_admin = 2;
   int64 ttl = 3 [(gogoproto.customname) = "TTL"];
-  ClusterRoles cluster_roles = 4;
 }
 
 //// Authorization data structures
 
-// Scope (actually a "role" in canonical security nomenclature) represents a
-// rough level of access that a principal has to a repo
-enum Scope {
-  // To remove a user's scope from a repo, set their scope to NONE
-  NONE = 0;
-  READER = 1;
-  WRITER = 2;
-  OWNER = 3;
+// Roles represents the set of roles a principal has
+message Roles {
+  map<string, bool> roles = 1;
+}
+
+// RoleBinding represents the set of roles principals have on a given Resource
+message RoleBinding {
```

Comment:
This is like, a very very minor nit, but I think the naming scheme here is slightly different at least from the one Kubernetes uses (I haven't checked GCP, actually. I can check that and update this comment). My understanding is that in Kubernetes, a RoleBinding is a single pair: `(principle, role)`, so if I have three roles on the same resource, that's technically three RoleBindings.

I think it's fine to keep this as-is, but I wanted to register the possible discrepency in case you think it might matter.

Reply:
I originally had `RoleBinding` pluralized, and then in some instances that changed a single binding I had it singular, and it made it harder to keep track of what APIs and structs had each spelling. I figured for simplicity it could just always be singular, even though this is really a collection of role bindings.